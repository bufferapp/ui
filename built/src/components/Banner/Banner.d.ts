import React from 'react';
export default class Banner extends React.Component {
    constructor(props: any);
    closeBanner(): void;
    renderBannerContent(themeColor: any): React.JSX.Element;
    render(): React.JSX.Element | null;
}
//# sourceMappingURL=Banner.d.ts.map