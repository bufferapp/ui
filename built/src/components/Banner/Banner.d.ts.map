{"version":3,"file":"Banner.d.ts","sourceRoot":"","sources":["../../../../src/components/Banner/Banner.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;AAezB,MAAM,CAAC,OAAO,OAAO,MAAO,SAAQ,KAAK,CAAC,SAAS;gBAErC,KAAK,KAAA;IAUjB,WAAW;IAUX,mBAAmB,CAAC,UAAU,KAAA;IA6B9B,MAAM;CA8BP"}