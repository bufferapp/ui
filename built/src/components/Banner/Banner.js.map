{"version":3,"file":"Banner.js","sourceRoot":"","sources":["../../../../src/components/Banner/Banner.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;AACzB,OAAO,SAAS,MAAM,YAAY,CAAA;AAClC,OAAO,MAAM,MAAM,WAAW,CAAA;AAC9B,OAAO,IAAI,MAAM,SAAS,CAAA;AAC1B,OAAO,SAAS,MAAM,qBAAqB,CAAA;AAE3C,OAAO,EACL,YAAY,EACZ,iBAAiB,EACjB,OAAO,EACP,aAAa,GACd,MAAM,SAAS,CAAA;AAEhB,OAAO,EAAE,YAAY,EAAE,MAAM,iBAAiB,CAAA;AAE9C,MAAM,CAAC,OAAO,OAAO,MAAO,SAAQ,KAAK,CAAC,SAAS;IAEjD,YAAY,KAAK;QACf,KAAK,CAAC,KAAK,CAAC,CAAA;QAEZ,IAAI,CAAC,KAAK,GAAG;YACX,MAAM,EAAE,IAAI;SACb,CAAA;QAED,IAAI,CAAC,WAAW,GAAG,IAAI,CAAC,WAAW,CAAC,IAAI,CAAC,IAAI,CAAC,CAAA;IAChD,CAAC;IAED,WAAW;QACT,IAAI,CAAC,QAAQ,CAAC,EAAE,MAAM,EAAE,KAAK,EAAE,CAAC,CAAA;QAEhC,MAAM,EAAE,aAAa,EAAE,GAAG,IAAI,CAAC,KAAK,CAAA;QACpC,IAAI,aAAa,EAAE;YACjB,aAAa,EAAE,CAAA;SAChB;IACH,CAAC;IAGD,mBAAmB,CAAC,UAAU;QAE5B,MAAM,EAAE,UAAU,EAAE,IAAI,EAAE,YAAY,EAAE,GAAG,IAAI,CAAC,KAAK,CAAA;QACrD,IAAI,UAAU,EAAE;YACd,OAAO,CACL,oBAAC,OAAO;gBAEN,6BAAK,uBAAuB,EAAE,UAAU,GAAI,CACpC,CACX,CAAA;SACF;QACD,OAAO,CACL,oBAAC,OAAO;YACN,oBAAC,IAAI,IAAC,IAAI,EAAC,WAAW,EAAC,KAAK,EAAC,MAAM,IAChC,IAAI,CACA;YACP,oBAAC,aAAa;gBAEZ,oBAAC,MAAM,IACL,IAAI,EAAE,UAAU,KAAK,QAAQ,CAAC,CAAC,CAAC,QAAQ,CAAC,CAAC,CAAC,SAAS,EACpD,OAAO,EAAE,YAAY,CAAC,MAAM,EAC5B,KAAK,EAAE,YAAY,CAAC,KAAK,EACzB,IAAI,EAAC,OAAO,GACZ,CACY,CACR,CACX,CAAA;IACH,CAAC;IAED,MAAM;QAEJ,MAAM,EAAE,MAAM,EAAE,GAAG,IAAI,CAAC,KAAK,CAAA;QAE7B,MAAM,EAAE,UAAU,EAAE,GAAG,IAAI,CAAC,KAAK,CAAA;QAEjC,IAAI,MAAM,EAAE;YACV,OAAO,CACL,oBAAC,YAAY,IAAC,UAAU,EAAE,UAAU;gBACjC,IAAI,CAAC,mBAAmB,CAAC,UAAU,CAAC;gBACrC,oBAAC,iBAAiB;oBAEhB,oBAAC,MAAM,IACL,IAAI,EAAC,MAAM,EACX,IAAI,EAAE,CACJ,oBAAC,SAAS,IACR,KAAK,EAAE,UAAU,KAAK,MAAM,CAAC,CAAC,CAAC,MAAM,CAAC,CAAC,CAAC,YAAY,GACpD,CACjB,EACa,WAAW,QACX,OAAO,EAAE,IAAI,CAAC,WAAW,EACzB,KAAK,EAAC,OAAO,EACb,IAAI,EAAC,OAAO,GACZ,CACgB,CACP,CAChB,CAAA;SACF;QACD,OAAO,IAAI,CAAA;IACb,CAAC;CACF;AAGD,MAAM,CAAC,SAAS,GAAG;IAEjB,IAAI,EAAE,SAAS,CAAC,MAAM;IAGtB,YAAY,EAAE,SAAS,CAAC,KAAK,CAAC;QAC5B,KAAK,EAAE,SAAS,CAAC,MAAM;QACvB,MAAM,EAAE,SAAS,CAAC,IAAI;KACvB,CAAC;IAGF,UAAU,EAAE,SAAS,CAAC,KAAK,CAAC,EAAE,MAAM,EAAE,SAAS,CAAC,MAAM,EAAE,CAAC;IAGzD,UAAU,EAAE,SAAS,CAAC,KAAK,CAAC,CAAC,MAAM,EAAE,QAAQ,CAAC,CAAC;IAG/C,aAAa,EAAE,SAAS,CAAC,IAAI;CAC9B,CAAA;AAGD,MAAM,CAAC,YAAY,GAAG;IACpB,IAAI,EAAE,EAAE;IACR,YAAY,EAAE,EAAE;IAChB,UAAU,EAAE,IAAI;IAChB,UAAU,EAAE,MAAM;IAClB,aAAa,EAAE,IAAI;CACpB,CAAA","sourcesContent":["import React from 'react'\nimport PropTypes from 'prop-types'\nimport Button from '../Button'\nimport Text from '../Text'\nimport CrossIcon from '../Icon/Icons/Cross'\n\nimport {\n  BannerStyled,\n  BannerCloseButton,\n  Wrapper,\n  ButtonWrapper,\n} from './style'\n\nimport { orangeDarker } from '../style/colors'\n\nexport default class Banner extends React.Component {\n  // @ts-expect-error TS(7006) FIXME: Parameter 'props' implicitly has an 'any' type.\n  constructor(props) {\n    super(props)\n\n    this.state = {\n      isOpen: true,\n    }\n\n    this.closeBanner = this.closeBanner.bind(this)\n  }\n\n  closeBanner() {\n    this.setState({ isOpen: false })\n    // @ts-expect-error TS(2339) FIXME: Property 'onCloseBanner' does not exist on type 'R... Remove this comment to see the full error message\n    const { onCloseBanner } = this.props\n    if (onCloseBanner) {\n      onCloseBanner()\n    }\n  }\n\n  // @ts-expect-error TS(7006) FIXME: Parameter 'themeColor' implicitly has an 'any' typ... Remove this comment to see the full error message\n  renderBannerContent(themeColor) {\n    // @ts-expect-error TS(2339) FIXME: Property 'customHTML' does not exist on type 'Read... Remove this comment to see the full error message\n    const { customHTML, text, actionButton } = this.props\n    if (customHTML) {\n      return (\n        <Wrapper>\n          {/* eslint-disable-next-line */}\n          <div dangerouslySetInnerHTML={customHTML} />\n        </Wrapper>\n      )\n    }\n    return (\n      <Wrapper>\n        <Text type=\"paragraph\" color=\"#FFF\">\n          {text}\n        </Text>\n        <ButtonWrapper>\n          {/* @ts-expect-error TS(2740) FIXME: Type '{ type: string; onClick: any; label: any; si... Remove this comment to see the full error message */}\n          <Button\n            type={themeColor === 'orange' ? 'orange' : 'primary'}\n            onClick={actionButton.action}\n            label={actionButton.label}\n            size=\"small\"\n          />\n        </ButtonWrapper>\n      </Wrapper>\n    )\n  }\n\n  render() {\n    // @ts-expect-error TS(2339) FIXME: Property 'isOpen' does not exist on type 'Readonly... Remove this comment to see the full error message\n    const { isOpen } = this.state\n    // @ts-expect-error TS(2339) FIXME: Property 'themeColor' does not exist on type 'Read... Remove this comment to see the full error message\n    const { themeColor } = this.props\n\n    if (isOpen) {\n      return (\n        <BannerStyled themeColor={themeColor}>\n          {this.renderBannerContent(themeColor)}\n          <BannerCloseButton>\n            {/* @ts-expect-error TS(2740) FIXME: Type '{ type: string; icon: Element; hasIconOnly: ... Remove this comment to see the full error message */}\n            <Button\n              type=\"text\"\n              icon={(\n                <CrossIcon\n                  color={themeColor === 'blue' ? '#fff' : orangeDarker}\n                />\n)}\n              hasIconOnly\n              onClick={this.closeBanner}\n              label=\"Close\"\n              size=\"small\"\n            />\n          </BannerCloseButton>\n        </BannerStyled>\n      )\n    }\n    return null\n  }\n}\n\n// @ts-expect-error TS(2339) FIXME: Property 'propTypes' does not exist on type 'typeo... Remove this comment to see the full error message\nBanner.propTypes = {\n  /** The main text of the banner */\n  text: PropTypes.string,\n\n  /** The text of the Call To Action of the banner */\n  actionButton: PropTypes.shape({\n    label: PropTypes.string,\n    action: PropTypes.func,\n  }),\n\n  /** Custom HTML */\n  customHTML: PropTypes.shape({ __html: PropTypes.string }),\n\n  /** Theme color. Can be `'blue'` or `'orange'` */\n  themeColor: PropTypes.oneOf(['blue', 'orange']),\n\n  /** Handler when the banner closes */\n  onCloseBanner: PropTypes.func,\n}\n\n// @ts-expect-error TS(2339) FIXME: Property 'defaultProps' does not exist on type 'ty... Remove this comment to see the full error message\nBanner.defaultProps = {\n  text: '',\n  actionButton: {},\n  customHTML: null,\n  themeColor: 'blue',\n  onCloseBanner: null,\n}\n"]}