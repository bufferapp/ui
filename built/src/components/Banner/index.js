export { default } from './Banner';
//# sourceMappingURL=index.js.map