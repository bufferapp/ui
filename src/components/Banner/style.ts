import styled from 'styled-components'
import { blueDarker, orangeLighter, orangeDarker, white } from '../style/colors'

import { fontSize, fontFamily } from '../style/fonts'

export const BannerStyled = styled.div`
  align-items: center;
  background: ${(props) =>
    // @ts-expect-error TS(2339) FIXME: Property 'themeColor' does not exist on type 'Them... Remove this comment to see the full error message
    props.themeColor === 'blue' ? blueDarker : orangeLighter};
  box-sizing: border-box;
  color: ${(props): string =>
    // @ts-expect-error TS(2339) FIXME: Property 'themeColor' does not exist on type 'Them... Remove this comment to see the full error message
    props.themeColor === 'blue' ? white : orangeDarker};
  display: flex;
  font-family: ${fontFamily};
  font-size: ${fontSize};
  height: 48px;
  padding-left: 8px;
  padding-right: 32px;
  position: relative;
  width: 100vw;

  /** Style inline links */
  a {
    color: ${(props): string =>
      // @ts-expect-error TS(2339) FIXME: Property 'themeColor' does not exist on type 'Them... Remove this comment to see the full error message
      props.themeColor === 'blue' ? white : orangeDarker};
  }
`

export const BannerCloseButton = styled.div`
  position: absolute;
  right: 8px;
`

export const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex: 1 1 auto;
`

export const ButtonWrapper = styled.div`
  margin-left: 16px;
`
