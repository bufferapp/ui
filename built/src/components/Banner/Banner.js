import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';
import Text from '../Text';
import CrossIcon from '../Icon/Icons/Cross';
import { BannerStyled, BannerCloseButton, Wrapper, ButtonWrapper, } from './style';
import { orangeDarker } from '../style/colors';
export default class Banner extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isOpen: true,
        };
        this.closeBanner = this.closeBanner.bind(this);
    }
    closeBanner() {
        this.setState({ isOpen: false });
        const { onCloseBanner } = this.props;
        if (onCloseBanner) {
            onCloseBanner();
        }
    }
    renderBannerContent(themeColor) {
        const { customHTML, text, actionButton } = this.props;
        if (customHTML) {
            return (React.createElement(Wrapper, null,
                React.createElement("div", { dangerouslySetInnerHTML: customHTML })));
        }
        return (React.createElement(Wrapper, null,
            React.createElement(Text, { type: "paragraph", color: "#FFF" }, text),
            React.createElement(ButtonWrapper, null,
                React.createElement(Button, { type: themeColor === 'orange' ? 'orange' : 'primary', onClick: actionButton.action, label: actionButton.label, size: "small" }))));
    }
    render() {
        const { isOpen } = this.state;
        const { themeColor } = this.props;
        if (isOpen) {
            return (React.createElement(BannerStyled, { themeColor: themeColor },
                this.renderBannerContent(themeColor),
                React.createElement(BannerCloseButton, null,
                    React.createElement(Button, { type: "text", icon: (React.createElement(CrossIcon, { color: themeColor === 'blue' ? '#fff' : orangeDarker })), hasIconOnly: true, onClick: this.closeBanner, label: "Close", size: "small" }))));
        }
        return null;
    }
}
Banner.propTypes = {
    text: PropTypes.string,
    actionButton: PropTypes.shape({
        label: PropTypes.string,
        action: PropTypes.func,
    }),
    customHTML: PropTypes.shape({ __html: PropTypes.string }),
    themeColor: PropTypes.oneOf(['blue', 'orange']),
    onCloseBanner: PropTypes.func,
};
Banner.defaultProps = {
    text: '',
    actionButton: {},
    customHTML: null,
    themeColor: 'blue',
    onCloseBanner: null,
};
//# sourceMappingURL=Banner.js.map