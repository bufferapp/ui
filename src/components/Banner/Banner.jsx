import React from 'react'
import PropTypes from 'prop-types'
import Button from '../Button'
import Text from '../Text'
import CrossIcon from '../Icon/Icons/Cross'

import {
  BannerStyled,
  BannerCloseButton,
  Wrapper,
  ButtonWrapper,
} from './style'

import { orangeDarker } from '../style/colors'

export default class Banner extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      isOpen: true,
    }

    this.closeBanner = this.closeBanner.bind(this)
  }

  closeBanner() {
    this.setState({ isOpen: false })
    const { onCloseBanner } = this.props
    if (onCloseBanner) {
      onCloseBanner()
    }
  }

  renderBannerContent(themeColor) {
    const { customHTML, text, actionButton } = this.props
    if (customHTML) {
      return (
        <Wrapper>
          {/* eslint-disable-next-line */}
          <div dangerouslySetInnerHTML={customHTML} />
        </Wrapper>
      )
    }
    return (
      <Wrapper>
        <Text type="paragraph" color="#FFF">
          {text}
        </Text>
        <ButtonWrapper>
          <Button
            type={themeColor === 'orange' ? 'orange' : 'primary'}
            onClick={actionButton.action}
            label={actionButton.label}
            size="small"
          />
        </ButtonWrapper>
      </Wrapper>
    )
  }

  render() {
    const { isOpen } = this.state
    const { themeColor } = this.props

    if (isOpen) {
      return (
        <BannerStyled themeColor={themeColor}>
          {this.renderBannerContent(themeColor)}
          <BannerCloseButton>
            <Button
              type="text"
              icon={
                <CrossIcon
                  color={themeColor === 'blue' ? '#fff' : orangeDarker}
                />
              }
              hasIconOnly
              onClick={this.closeBanner}
              label="Close"
              size="small"
            />
          </BannerCloseButton>
        </BannerStyled>
      )
    }
    return null
  }
}

Banner.propTypes = {
  /** The main text of the banner */
  text: PropTypes.string,

  /** The text of the Call To Action of the banner */
  actionButton: PropTypes.shape({
    label: PropTypes.string,
    action: PropTypes.func,
  }),

  /** Custom HTML */
  customHTML: PropTypes.shape({ __html: PropTypes.string }),

  /** Theme color. Can be `'blue'` or `'orange'` */
  themeColor: PropTypes.oneOf(['blue', 'orange']),

  /** Handler when the banner closes */
  onCloseBanner: PropTypes.func,
}

Banner.defaultProps = {
  text: '',
  actionButton: {},
  customHTML: null,
  themeColor: 'blue',
  onCloseBanner: null,
}
