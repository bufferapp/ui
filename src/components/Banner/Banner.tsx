import React from 'react'
import PropTypes from 'prop-types'
import Button from '../Button'
import Text from '../Text'
import CrossIcon from '../Icon/Icons/Cross'

import {
  BannerStyled,
  BannerCloseButton,
  Wrapper,
  ButtonWrapper,
} from './style'

import { orangeDarker } from '../style/colors'

export default class Banner extends React.Component {
  // @ts-expect-error TS(7006) FIXME: Parameter 'props' implicitly has an 'any' type.
  constructor(props) {
    super(props)

    this.state = {
      isOpen: true,
    }

    this.closeBanner = this.closeBanner.bind(this)
  }

  closeBanner() {
    this.setState({ isOpen: false })
    // @ts-expect-error TS(2339) FIXME: Property 'onCloseBanner' does not exist on type 'R... Remove this comment to see the full error message
    const { onCloseBanner } = this.props
    if (onCloseBanner) {
      onCloseBanner()
    }
  }

  // @ts-expect-error TS(7006) FIXME: Parameter 'themeColor' implicitly has an 'any' typ... Remove this comment to see the full error message
  renderBannerContent(themeColor) {
    // @ts-expect-error TS(2339) FIXME: Property 'customHTML' does not exist on type 'Read... Remove this comment to see the full error message
    const { customHTML, text, actionButton } = this.props
    if (customHTML) {
      return (
        <Wrapper>
          {/* eslint-disable-next-line */}
          <div dangerouslySetInnerHTML={customHTML} />
        </Wrapper>
      )
    }
    return (
      <Wrapper>
        <Text type="paragraph" color="#FFF">
          {text}
        </Text>
        <ButtonWrapper>
          {/* @ts-expect-error TS(2740) FIXME: Type '{ type: string; onClick: any; label: any; si... Remove this comment to see the full error message */}
          <Button
            type={themeColor === 'orange' ? 'orange' : 'primary'}
            onClick={actionButton.action}
            label={actionButton.label}
            size="small"
          />
        </ButtonWrapper>
      </Wrapper>
    )
  }

  render() {
    // @ts-expect-error TS(2339) FIXME: Property 'isOpen' does not exist on type 'Readonly... Remove this comment to see the full error message
    const { isOpen } = this.state
    // @ts-expect-error TS(2339) FIXME: Property 'themeColor' does not exist on type 'Read... Remove this comment to see the full error message
    const { themeColor } = this.props

    if (isOpen) {
      return (
        // @ts-expect-error TS(2769) FIXME: No overload matches this call.
        <BannerStyled themeColor={themeColor}>
          {this.renderBannerContent(themeColor)}
          <BannerCloseButton>
            {/* @ts-expect-error TS(2740) FIXME: Type '{ type: string; icon: Element; hasIconOnly: ... Remove this comment to see the full error message */}
            <Button
              type="text"
              icon={
                <CrossIcon
                  color={themeColor === 'blue' ? '#fff' : orangeDarker}
                />
              }
              hasIconOnly
              onClick={this.closeBanner}
              label="Close"
              size="small"
            />
          </BannerCloseButton>
        </BannerStyled>
      )
    }
    return null
  }
}

// @ts-expect-error TS(2339) FIXME: Property 'propTypes' does not exist on type 'typeo... Remove this comment to see the full error message
Banner.propTypes = {
  /** The main text of the banner */
  text: PropTypes.string,

  /** The text of the Call To Action of the banner */
  actionButton: PropTypes.shape({
    label: PropTypes.string,
    action: PropTypes.func,
  }),

  /** Custom HTML */
  customHTML: PropTypes.shape({ __html: PropTypes.string }),

  /** Theme color. Can be `'blue'` or `'orange'` */
  themeColor: PropTypes.oneOf(['blue', 'orange']),

  /** Handler when the banner closes */
  onCloseBanner: PropTypes.func,
}

// @ts-expect-error TS(2339) FIXME: Property 'defaultProps' does not exist on type 'ty... Remove this comment to see the full error message
Banner.defaultProps = {
  text: '',
  actionButton: {},
  customHTML: null,
  themeColor: 'blue',
  onCloseBanner: null,
}
