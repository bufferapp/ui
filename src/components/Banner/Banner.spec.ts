/* eslint-disable react/jsx-filename-extension */
// @ts-expect-error TS(7016) FIXME: Could not find a declaration file for module 'jest... Remove this comment to see the full error message
import snap from 'jest-auto-snapshots'
import 'jest-styled-components'

import Banner from './Banner'

snap(Banner, './Banner.tsx')
