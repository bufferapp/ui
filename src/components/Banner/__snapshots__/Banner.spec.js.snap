// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > Banner Matches snapshot when passed all props 1`] = `
<ForwardRef(styled.div)
  themeColor="blue"
>
  <ForwardRef(styled.div)>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "jest-auto-snapshots String Fixture",
        }
      }
    />
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)>
    <Button
      disabled={false}
      fullWidth={false}
      hasIconOnly={true}
      hideSearch={true}
      icon={
        <CrossIcon
          color="#fff"
        />
      }
      iconEnd={false}
      label="Close"
      loading={false}
      onClick={[Function]}
      onOpen={null}
      selectPosition="bottom"
      size="small"
      type="text"
    />
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;
